packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -109,6 +129,35 @@ export class PeerExchangeDiscovery
     return "@waku/peer-exchange";
   }
 
+  private readonly startRecurringQueries = async (
+    peerId: PeerId,
+    firstQueryTimeout: number = DEFAULT_PEER_EXCHANGE_INITIAL_QUERY_TIMEOUT_MS,
+    attemptsBeforeBackoff: number = DEFAULT_ATTEMPTS_BEFORE_BACKOFF,
+    queryInterval: number = DEFAULT_PEER_EXCHANGE_QUERY_INTERVAL_MS
+  ): Promise<void> => {
+    const peerIdStr = peerId.toString();
+    if (!this.queryAttempts.has(peerIdStr))
+      this.queryAttempts.set(peerIdStr, 1);
+
+    setTimeout(async () => {
+      await this.query(peerId);
+    }, firstQueryTimeout);
+
+    const interval = setInterval(async () => {
+      const currentAttempt = this.queryAttempts.get(peerIdStr) as number;
```

Comment:
```suggestion
      const currentAttempt = this.queryAttempts.get(peerIdStr) ?? 1;
```

And you can remove

```
    if (!this.queryAttempts.has(peerIdStr))
      this.queryAttempts.set(peerIdStr, 1);
```

above.